app/src/main/java/de/danoeh/antennapod/activity/MediaplayerActivity.java

```diff
@@ -677,8 +678,13 @@ void onPositionObserverUpdate() {
         if (controller == null || txtvPosition == null || txtvLength == null) {
             return;
         }
-        int currentPosition = controller.getPosition();
-        int duration = controller.getDuration();
+
+        TimeSpeedConverter timeSpeedConverter = new TimeSpeedConverter(
+                UserPreferences.isTimeRespectSpeed(),
+                Float.parseFloat(UserPreferences.getPlaybackSpeed()));
+        int currentPosition = timeSpeedConverter.convert(controller.getPosition());
+        int duration = timeSpeedConverter.convert(controller.getDuration());
+        int remainingTime = timeSpeedConverter.convert(controller.getDuration());
```

Comment:
Shouldn't the remaining time be `duration - currentPosition`? Currently, it looks like it is just a copy of the duration

Reply:
oups, my apologies, I think I messed up a rebase.
Thanks for catching this